src/components/AddPost/AddPost.jsx

```diff
@@ -0,0 +1,164 @@
+import Picker from "emoji-picker-react";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useSelector, useDispatch } from "react-redux";
+
+import { addNewPost, fetchAllPosts } from "../../firebase/firestore-requests";
+import { addUserPost } from "features/user/userSlice";
+import {
+  Stack,
+  Button,
+  Box,
+  Avatar,
+  TextField,
+  IconButton,
+} from "@mui/material";
+
+import PhotoCameraBackIcon from "@mui/icons-material/PhotoCameraBack";
+import EmojiEmotionsIcon from "@mui/icons-material/EmojiEmotions";
```

Comment:
suggestion - material UI imports can be moved to a common utilities folder

Reply:
thanks for the suggestion, I'll sure abstract them later